### PR TITLE
feat(FEC-9220): QoE tidbits

### PR DESCRIPTION
**Pull request by RoyBregman**
### Description of the Changes

changing soundMode, tabMode and networkConnectionType - to be updated in model upon event change

Solves FEC-9220
### CheckLists

- [ ] changes have been done against master branch, and PR does not conflict
- [ ] new unit / functional tests have been added (whenever applicable)
- [ ] test are passing in local environment 
- [ ] Travis tests are passing (or test results are not worse than on master branch :))
- [ ] Docs have been updated

### CODE REVIEW

File: src/kava.js [hunk 1]
```diff
@@ -445,13 +450,28 @@ class Kava extends BasePlugin {
...
+    this._model.updateModel({
+      networkConnectionType: this._navConnection.effectiveType
+    });
+  }
+
+  _initNetworkConnectionType(): void {
+    this._navConnection = window.navigator.connection || window.navigator.mozConnection || window.navigator.webkitConnection;
+
+    if (this._navConnection) {
+      this._navConnection.addEventListener('change', this._updateNetworkConnectionTypeinModel.bind(this));
```
Comment (yairans) [inline src/kava.js]: use `eventManager ` for cleaning

File: src/kava.js [hunk 2]
```diff
@@ -87,6 +90,15 @@ class Kava extends BasePlugin {
         this._model.updateModel({playerJSLoadTime: entry.duration});
       }
     }
+
+    this._initTabMode();
```
Comment (yairans) [inline src/kava.js]: move to `addBinding`

File: src/kava.js [hunk 3]
```diff
@@ -87,6 +90,15 @@ class Kava extends BasePlugin {
...
+    this._initNetworkConnectionType();
```
Comment (yairans) [inline src/kava.js]: same

File: src/kava.js [hunk 4]
```diff
@@ -746,20 +766,30 @@ class Kava extends BasePlugin {
...
     } else if (typeof document.webkitHidden !== 'undefined') {
-      hidden = 'webkitHidden';
-    } else {
-      return false;
+      this._hiddenAttr = 'webkitHidden';
+      this._visibilityChangeEventName = 'webkitvisibilitychange';
+    }
+
+    if (this._hiddenAttr !== 'undefined') {
+      document.addEventListener(this._visibilityChangeEventName, this._updateTabModeinModel.bind(this));
```
Comment (yairans) [inline src/kava.js]: use `eventManager`

File: src/kava.js [hunk 5]
```diff
@@ -746,20 +766,30 @@ class Kava extends BasePlugin {
...
+      tabMode: document[this._hiddenAttr] ? TabMode.TAB_NOT_FOCUSED : TabMode.TAB_FOCUSED
+    });
+  }
+
+  _initTabMode(): void {
     if (typeof document.hidden !== 'undefined') {
       // Opera 12.10 and Firefox 18 and later support
-      hidden = 'hidden';
+      this._hiddenAttr = 'hidden';
+      this._visibilityChangeEventName = 'visibilitychange';
```
Comment (yairans) [inline src/kava.js]: move to `const`

File: src/kava.js [hunk 6]
```diff
@@ -746,20 +766,30 @@ class Kava extends BasePlugin {
...
+      // $FlowFixMe
+      tabMode: document[this._hiddenAttr] ? TabMode.TAB_NOT_FOCUSED : TabMode.TAB_FOCUSED
+    });
+  }
+
+  _initTabMode(): void {
     if (typeof document.hidden !== 'undefined') {
       // Opera 12.10 and Firefox 18 and later support
-      hidden = 'hidden';
+      this._hiddenAttr = 'hidden';
```
Comment (yairans) [inline src/kava.js]: move to `const` and send to `_updateTabModeinModel` as a parameter

File: src/kava.js [hunk 7]
```diff
@@ -37,6 +37,9 @@ class Kava extends BasePlugin {
   _performanceEntries: window.PerformanceEntry[] = [];
   _pendingFragLoadedUrls: string[] = [];
   _fragLoadedFiredOnce: boolean = false;
+  _hiddenAttr: string;
+  _visibilityChangeEventName: string;
```
Comment (yairans) [inline src/kava.js]: remove after moving to const

File: src/kava.js (hunk 1 above)
Comment (RoyBregman) [reply src/kava.js]: done

File: src/kava.js (hunk 2 above)
Comment (RoyBregman) [reply src/kava.js]: done

File: src/kava.js (hunk 3 above)
Comment (RoyBregman) [reply src/kava.js]: done

File: src/kava.js (hunk 6 above)
Comment (RoyBregman) [reply src/kava.js]: done. but as let and not const as it is assigned not upon declaration and fails in eslint

File: src/kava.js [hunk 8]
```diff
@@ -746,20 +762,34 @@ class Kava extends BasePlugin {
...
     } else if (typeof document.webkitHidden !== 'undefined') {
-      hidden = 'webkitHidden';
-    } else {
-      return false;
+      hiddenAttr = 'webkitHidden';
+      visibilityChangeEventName = 'webkitvisibilitychange';
+    }
+
+    if (hiddenAttr && visibilityChangeEventName) {
+      document.addEventListener(visibilityChangeEventName, () => {
```
Comment (yairans) [inline src/kava.js]: use `eventManager`

**Pull request merged**